### DISCUSSION TITLE
filterOptions doesn't work with async function

### DISCUSSION THREAD

**Discussion by gubarez** [payloadcms/payload#1975] [Feature Requests & Ideas]
# Bug Report

In relationship, filterOptions doesn't use the returned value of the async function.

## Steps to Reproduce

```
{
            name: 'variations',
            type: 'relationship',
            relationTo: 'product-variations',
            hasMany: true,
            unique: true,
            filterOptions: async () => ({
                myfield: {equals: 'value1'}
            }),
}
```

## Other Details

Using payload version 1.5.6

**Comment by JarrodMFlesch** [comment]
@gubarez this is currently not supported. You can see the [type for filterOptions](https://github.com/payloadcms/payload/blob/master/src/fields/config/types.ts#L57) does not return a promise and therefore does not support async.

[This](https://github.com/payloadcms/payload/blob/master/src/admin/components/forms/field-types/getFilterOptionsQuery.ts#L10) file would need to be updated, and where `getFilterOptionsQuery` is called to support async/await. If you want to throw together a PR for this support let me know! 

For now I will convert this into a discussion 🙂
